pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -333,8 +352,19 @@ func (ssr *Snapshotter) TakeDeltaSnapshot() (*snapstore.Snapshot, error) {
 		return nil, fmt.Errorf("failed to compute hash of events: %v", err)
 	}
 	ssr.events = hash.Sum(ssr.events)
+
 	startTime := time.Now()
-	if err := ssr.store.Save(*snap, ioutil.NopCloser(bytes.NewReader(ssr.events))); err != nil {
+	rc := ioutil.NopCloser(bytes.NewReader(ssr.events))
+
+	// if compression is enabled
+	//    then compress the snapshot.
+	if ssr.compressionConfig.Enabled {
+		ssr.logger.Info("start the Compression of delta snapshot")
+		rc = compressor.CompressSnapshot(rc, ssr.compressionConfig.CompressionPolicy)
```

Comment:
If `CompressSnapshot` returns error (if you add the default case) then it will have to be handled here.